### PR TITLE
Add: print out the stack

### PR DESCRIPTION
**Pull request by Jim8y**
This pr override the ToString() to the EvaluationStack.

WIth this pr, we can print the stack while debuging the vm.

### CODE REVIEW

File: src/Neo.VM/EvaluationStack.cs [hunk 1]
```diff
@@ -35,6 +35,23 @@ internal EvaluationStack(ReferenceCounter referenceCounter)
...
+        public override string ToString()
+        {
+            var reversed = new Stack<StackItem>(innerList);
+            return $@"[{string.Join(", ", reversed.Select(p =>
+                    {
+                        var value = p.Type switch
+                        {
+                            StackItemType.Boolean => $"({p.GetBoolean()})",
+                            StackItemType.Integer => $"({p.GetInteger()})",
+                            StackItemType.ByteString => $"({p.GetString()})",
```
Comment (shargon) [inline src/Neo.VM/EvaluationStack.cs:47]:
```suggestion
                            StackItemType.ByteString => $"(\"{p.GetString()}\")",
```

File: tests/Neo.VM.Tests/UtEvaluationStack.cs [hunk 2]
```diff
@@ -187,5 +187,18 @@ public void TestReverse()
...
+        public void TestToString()
+        {
+            var stack = new EvaluationStack(new ReferenceCounter());
+
+            stack.Insert(0, 3);
+            stack.Insert(1, 1);
+            stack.Insert(2, "test");
+            stack.Insert(3, true);
+
+            Assert.AreEqual("[Integer(3), Integer(1), ByteString(test), Boolean(True)]", stack.ToString());
```
Comment (shargon) [inline tests/Neo.VM.Tests/UtEvaluationStack.cs:201]:
```suggestion
            Assert.AreEqual("[Integer(3), Integer(1), ByteString(\"test\"), Boolean(True)]", stack.ToString());
```

**Review by shargon — COMMENTED** [review]
We can use `DebuggerDisplayAttribute` ?

**Comment by Liaojinghui** [comment]
Screenshot 2023-12-19 at 8 25 28 AM

i am expecting something like this, can it be achieved with "DebuggerDisplayAttribute"?

**Comment by shargon** [comment]
`DebuggerDisplayAttribute` is only for show the information while the cursor is above during the debug

File: src/Neo.VM/EvaluationStack.cs [hunk 3]
```diff
@@ -35,6 +35,22 @@ internal EvaluationStack(ReferenceCounter referenceCounter)
...
+        public override string ToString()
+        {
+            return $@"[{string.Join(", ", innerList.Select(p =>
+                    {
+                        var value = p.Type switch
+                        {
+                            StackItemType.Boolean => $"({p.GetBoolean()})",
+                            StackItemType.Integer => $"({p.GetInteger()})",
+                            StackItemType.ByteString => $"(\"{p.GetString()}\")",
+                            _ => ""
```
Comment (shargon) [inline src/Neo.VM/EvaluationStack.cs:47]: CompundType => (p.Count) ?

Comment (Liaojinghui) [reply src/Neo.VM/EvaluationStack.cs:47]: Yes, if you think it is better

**Comment by Jim8y** [comment]
All set, please review

**Review by shargon — DISMISSED** [review]

File: src/Neo.VM/EvaluationStack.cs [hunk 4]
```diff
@@ -35,6 +35,25 @@ internal EvaluationStack(ReferenceCounter referenceCounter)
...
         public int Count => innerList.Count;
 
+        public override string ToString()
+        {
+            return $@"[{string.Join(", ", innerList.Select(p =>
+                    {
+                        var value = p.Type switch
+                        {
+                            StackItemType.Boolean => $"({p.GetBoolean()})",
+                            StackItemType.Integer => $"({p.GetInteger()})",
```
Comment (shargon) [inline src/Neo.VM/EvaluationStack.cs:45]:
```suggestion
                            StackItemType.Pointer => $"({((Pointer)ptr).Position})",
                            StackItemType.Boolean => $"({p.GetBoolean()})",
                            StackItemType.Integer => $"({p.GetInteger()})",
```

**Review by shargon — APPROVED** [review]

**Pull request merged**